pinot-spi/src/main/java/org/apache/pinot/spi/stream/PartitionGroupConsumer.java

```diff
@@ -55,15 +56,15 @@ default void start(StreamPartitionMsgOffset startOffset) {
    * @throws TimeoutException If the operation could not be completed within timeout
    * @return A batch of messages from the stream partition group
    */
-  default MessageBatch fetchMessages(StreamPartitionMsgOffset startOffset, int timeoutMs)
+  default MessageBatch fetchMessages(StreamPartitionMsgOffset startOffset, CommonConstants.Segment.Realtime.StreamContinuationMode continuationMode, int timeoutMs)
```

Comment:
I feel a boolean is easier to understand, and I don't see other possible mode

Reply:
Please also update the javadoc

Reply:
(MAJOR) This is backward incompatible (this is a public facing interface). We need to add default impl for the new added API and not changing existing signature